### ISSUE TITLE
Lint/AmbiguousOperator registers offense for macro methods

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#12212]
## Is your feature request related to a problem? Please describe.

See #4294 which was closed as stale.

RuboCop should not register an offence when passing a splat to macro methods.

```rb
class Foo
  attr_reader *BAR
end
```

Code like this is also not syntactically valid when adding a space to the right of `*`, which the cop suggests as a possible alternative:

```rb
class Foo
  delegate *BAR, to: :collection
end
```

## Describe the solution you'd like

A config option to mark methods to be ignored by this cop.

**Comment by koic** [comment]
This looks like the expected behavior to emulate the following Ruby's warning:

```console
$ ruby -vwce 'do_something *rest'
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-darwin19]
-e:1: warning: `*' interpreted as argument prefix
Syntax OK
```

So, I'll close this issue. Thank you.
